includes/classes/order.php

```diff
@@ -687,6 +691,8 @@ public function do_calculate_taxes($taxRates, $index, $taxCountryId, $taxZoneId)
               $this->info['tax_groups'][$taxDescription] = $taxAdd;
           }
       }
+      
+      $this->notify('NOTIFY_ORDER_CART_TAX_CALCULATIONS_END', array('index' => $index, 'tax_add' => $tax_add, 'shown_price' => $shown_price, 'taxRates' => $taxRates));
```

Comment:
Yes, at this point in the processing, the data is _informational_ only; the observer can use the information (specifically the `index`) to make any updates needed to the associated product's tax.